### PR TITLE
docs: transocder route config

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <zlizan@google.com>

#3456 

*Description*:
*Risk Level*: Low (docs only)
*Testing*: N/A
*Docs Changes*:
*Release Notes*:

### CODE REVIEW

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 1]
```diff
@@ -35,3 +35,30 @@ Then run protoc to generate the descriptor set from bookstore.proto:
     --descriptor_set_out=proto.pb test/proto/bookstore.proto
 
 If you have more than one proto source files, you can pass all of them in one command.
+
+Route config for transcoded requests
+------------------------------------
+
+The route config to be used with the gRPC-JSON transcoder should be identical with gRPC route,
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: s/identical with/identical to the

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 2]
```diff
@@ -35,3 +35,30 @@ Then run protoc to generate the descriptor set from bookstore.proto:
...
+The requests processed by the transcoder filter will have `/<package>.<service>/<method>` path and
+`POST` method. The route config for those request should match on `/<package>.<service>/<method>`
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: s/route config/route configs
s/request/requests

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 3]
```diff
@@ -35,3 +35,30 @@ Then run protoc to generate the descriptor set from bookstore.proto:
...
+but not the incoming request path. This allows the routes can be used for both gRPC requests and
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: s/but not/, not
s/can be used/to be used

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 4]
```diff
@@ -35,3 +35,30 @@ Then run protoc to generate the descriptor set from bookstore.proto:
...
+gRPC-JSON transcoded requests.
+
+For example, with the following proto example, the router will process `/helloworld.Greeter/SayHello`
+as path, so the route config prefix `/say` won't match requests to `SayHello`. If you want to match
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: s/as path/as the path

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 5]
```diff
@@ -35,3 +35,30 @@ Then run protoc to generate the descriptor set from bookstore.proto:
...
+incoming request path, set `match_incoming_request_route` to true.
```
Comment (mattklein123) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: the incoming

**Comment by lizan** [comment]
Done, thanks!

**Pull request merged**